packages/editor-ui/src/components/ValueSurvey.vue

```diff
@@ -72,67 +88,118 @@ export default mixins(workflowHelpers).extend({
 				email: '',
 				value: '',
 			},
+			loading: false,
 			showButtons: true,
 			VALUE_SURVEY_MODAL_KEY,
 		};
 	},
 	methods: {
 		closeDialog(): void {
 			if (this.form.value === '') {
-				this.$telemetry.track('User responded value survey score', { instance_id: this.$store.getters.instanceId, how_disappointed: '' });
-			} else if (this.form.value !== '') {
-				this.$telemetry.track('User responded value survey email', { instance_id: this.$store.getters.instanceId, email: '' });
+				this.$telemetry.track('User responded value survey score', {
+					instance_id: this.$store.getters.instanceId,
+					nps: '',
+				});
+			} else {
+				this.$telemetry.track('User responded value survey email', {
+					instance_id: this.$store.getters.instanceId,
+					email: '',
+				});
 			}
 
 			this.$store.commit('ui/closeTopModal');
 		},
+		onInputChange(value: string) {
+			this.form.email = value;
+		},
 		selectSurveyValue(value: string) {
 			this.form.value = value;
 			this.showButtons = false;
-			this.$store.dispatch('settings/submitValueSurvey', {value: this.form.value});
-			this.$telemetry.track('User responded value survey score', { instance_id: this.$store.getters.instanceId, how_disappointed: this.form.value });
+			this.loading = true;
+			this.$store.dispatch('settings/submitValueSurvey', { value: this.form.value }).then((response: IN8nPromptResponse) => {
+				if (response.updated) {
+					this.$telemetry.track('User responded value survey score', {
+						instance_id: this.$store.getters.instanceId,
+						nps: this.form.value,
+					});
+				}
+				this.loading = false;
+			});
+
 		},
 		send(): void {
```

Comment:
ux change - we need to support enter key

Reply:
Done